src/components/tooltip/Tooltip.tsx

```diff
@@ -142,67 +133,82 @@ const Tooltip: React.FC<TooltipProps> = ({
     ],
   });
 
-  const children = <span>{otherProps.children}</span>;
-
   const isVisible =
     typeof visible !== 'undefined'
       ? visible
       : displayTrigger === 'hover'
         ? showHover
         : showClick;
 
+  const childElementWithReference = React.Children.map(children, (child) => {
+    if (React.isValidElement(child)) {
+      return React.cloneElement(child, {
+        ref: setReferenceElement
+      })
+    } else {
+      return <span ref={setReferenceElement}>{child}</span>
+    }
+  })
+
   return (
-    <div ref={ref} className={className ? className + '_wrapper' : null}>
-      <SpanStyled ref={setReferenceElement}>
```

Comment:
not sure if I understood well this big refactor
But are we removing a span wrapper? 
I think this was mainly added to enable tooltip display on disabled elements (ex a disabled button)

Reply:
Hm, okay.

Seems like it disabled doesn't work with this branch right now.

Reply:
Added back the wrapping element around children props.